app/src/main/java/com/keylesspalace/tusky/components/filters/EditFilterActivity.kt

```diff
@@ -0,0 +1,354 @@
+package com.keylesspalace.tusky.components.filters
+
+import android.content.Context
+import android.os.Bundle
+import android.widget.ArrayAdapter
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.size
+import androidx.lifecycle.lifecycleScope
+import at.connyduck.calladapter.networkresult.NetworkResult
+import at.connyduck.calladapter.networkresult.fold
+import com.google.android.material.chip.Chip
+import com.google.android.material.snackbar.Snackbar
+import com.google.android.material.switchmaterial.SwitchMaterial
+import com.keylesspalace.tusky.BaseActivity
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.appstore.EventHub
+import com.keylesspalace.tusky.databinding.ActivityEditFilterBinding
+import com.keylesspalace.tusky.databinding.DialogFilterBinding
+import com.keylesspalace.tusky.entity.Filter
+import com.keylesspalace.tusky.entity.FilterKeyword
+import com.keylesspalace.tusky.network.MastodonApi
+import com.keylesspalace.tusky.util.afterTextChanged
+import com.keylesspalace.tusky.util.viewBinding
+import kotlinx.coroutines.launch
+import retrofit2.HttpException
+import java.util.Date
+import javax.inject.Inject
+
+class EditFilterActivity : BaseActivity() {
```

Comment:
Would be nice to have ViewModels for this and  FiltersActivity so they can handle screen rotation without reloading